File: 0645. Set Mismatch/Solution0645.py
class Solution0645:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        def swap(lists,n1,n2):
            temp = lists[n1]
            lists[n1] = lists[n2]
            lists[n2] = temp
    
        n = len(nums)
        for i in range(n):
            while nums[i] != i+1 and nums[nums[i]-1] != nums[i]:
                swap(nums, i, nums[i] - 1)
                
        dup, los = -1, -1
        for i in range(n):
            if nums[i] != i+1:
                dup = nums[i]
                los = nums[i-1]+1 if i > 0 else 1
        return [dup, los]

File: 0645. Set Mismatch/Solution0645.cpp
#include<vector>

using namespace std;

class Solution0645 {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            while (nums[i] != i+1 && nums[nums[i]-1] != nums[i]) {
                swap(nums[i], nums[nums[i]-1]);
            }
        }
        int dup = -1, los = -1;
        for (int i = 0; i < n; i++) {
            if (nums[i] != i+1) {
                dup = nums[i];
                los = i == 0 ? 1 : (nums[i-1]+1);
            }
        }
        return {dup, los};
    }
    
    void swap(int &a, int &b) {
        int temp = a;
        a = b;
        b = temp;
    }
};